subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf54lx.h

```diff
@@ -0,0 +1,639 @@
+	HAL_RADIO_NRF54LX_TXEN_TXIDLE_TX_S8_DEFAULT_NO_HW_TIFS_NS
+
+#define HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_1M_US \
+	HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_1M_DEFAULT_NO_HW_TIFS_US
+#define HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_1M_NS \
+	HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_1M_DEFAULT_NO_HW_TIFS_NS
+
+#define HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_2M_US \
+	HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_2M_DEFAULT_NO_HW_TIFS_US
+#define HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_2M_NS \
+	HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_2M_DEFAULT_NO_HW_TIFS_NS
+
+#define HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_S2_US \
+	HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_S2_DEFAULT_NO_HW_TIFS_US
+#define HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_S2_NS \
+	HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_S2_DEFAULT_NO_HW_TIFS_NS
+
+#define HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_S8_US \
+	HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_S8_DEFAULT_NO_HW_TIFS_US
+#define HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_S8_NS \
+	HAL_RADIO_NRF54LX_RXEN_RXIDLE_RX_S8_DEFAULT_NO_HW_TIFS_NS
+#endif /* !CONFIG_BT_CTLR_TIFS_HW */
+#endif /* !CONFIG_BT_CTLR_RADIO_ENABLE_FAST */
+
+/* HAL abstraction of Radio bitfields */
+#define HAL_RADIO_INTENSET_DISABLED_Msk         RADIO_INTENSET00_DISABLED_Msk
+#define HAL_RADIO_SHORTS_TRX_END_DISABLE_Msk    RADIO_SHORTS_PHYEND_DISABLE_Msk
+#define HAL_RADIO_SHORTS_TRX_PHYEND_DISABLE_Msk RADIO_SHORTS_PHYEND_DISABLE_Msk
+
+/* HAL abstraction of Radio IRQ number */
+#define HAL_RADIO_IRQn                          RADIO_0_IRQn
+
+/* SoC specific NRF_RADIO power-on reset value. Refer to Product Specification,
+ * RADIO Registers section for the documented reset values.
+ *
+ * NOTE: Only implementation used values defined here.
+ *       In the future if MDK or nRFx header include these, use them instead.
+ */
+#define HAL_RADIO_RESET_VALUE_DFEMODE       0x00000000UL
+#define HAL_RADIO_RESET_VALUE_CTEINLINECONF 0x00002800UL
+
+static inline void hal_radio_reset(void)
+{
+	/* TODO: Add any required setup for each radio event
+	 */
+}
+
+static inline void hal_radio_stop(void)
+{
+	/* TODO: Add any required cleanup of actions taken in hal_radio_reset()
+	 */
+}
+
+static inline void hal_radio_ram_prio_setup(void)
+{
+	/* TODO */
+}
+
+static inline uint32_t hal_radio_phy_mode_get(uint8_t phy, uint8_t flags)
+{
+	uint32_t mode;
+
+	switch (phy) {
+	case BIT(0):
+	default:
+		mode = RADIO_MODE_MODE_Ble_1Mbit;
+		break;
+
+	case BIT(1):
+		mode = RADIO_MODE_MODE_Ble_2Mbit;
+		break;
+
+#if defined(CONFIG_BT_CTLR_PHY_CODED)
+	case BIT(2):
+		if (flags & 0x01) {
+			mode = RADIO_MODE_MODE_Ble_LR125Kbit;
+		} else {
+			mode = RADIO_MODE_MODE_Ble_LR500Kbit;
+		}
+		break;
+#endif /* CONFIG_BT_CTLR_PHY_CODED */
+	}
+
+	return mode;
+}
+
+static inline uint32_t hal_radio_tx_power_max_get(void)
+{
+	return RADIO_TXPOWER_TXPOWER_0dBm;
+}
+
+static inline uint32_t hal_radio_tx_power_min_get(void)
+{
+	return RADIO_TXPOWER_TXPOWER_Neg40dBm;
+}
+
+static inline uint32_t hal_radio_tx_power_floor(int8_t tx_power_lvl)
+{
+	if (tx_power_lvl >= (int8_t)RADIO_TXPOWER_TXPOWER_0dBm) {
```

Comment:
These values are not anymore 8bits 2's complement representations of the power in dBm, but a funny mapping. So -40dBm (0x130) is bigger than -20dBm (0x1) or even 0dBm (0x13) :(